### PR TITLE
filter items in the list

### LINKED ISSUE #6: 6. As a user, I want to filter my shopping list to make it easier to locate an item in the list.
## Summary
Allowing users to search through their list will help them cut down on clutter and focus on items they’re looking for.

## Acceptance criteria
- [x] A form is added to the top of the `List` view, above the shopping list
- [x] The form includes the following elements
	- A text field (with semantic `<label>`!) which narrows down the list as the user types
	- When there’s text in the field, some kind of button (e.g., an X) to clear the field. When the field is cleared, the list is reset to its unfiltered state.

## Notes
- You will know your `<label>` is correctly implemented if you click on the label and keyboard focus moves to the related input. Refer to [the MDN docs on `<label>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for more info.
- The filter text should match *any* part of the item name (e.g., it should _not_ only match from the start of the string)
- [This wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png) shows the filter field, but not it being used

### PR DESCRIPTION
**Pull request by snowsneakers**
## Description

This code adds the ability for the user to filter the item list. We used the useState hook to store the user input and the data is filtered if it includes the input text. If the input length  > 0, a button that clear the input appears. If clicked, the input is cleared and the list become unfiltered.

## Related Issue

Closes #6 

## Acceptance Criteria

- [x] A form is added to the top of the `List` view, above the shopping list
- [x] The form includes the following elements
	- A text field (with semantic `<label>`!) which narrows down the list as the user types
	- When there’s text in the field, some kind of button (e.g., an X) to clear the field. When the field is cleared, the list is reset to its unfiltered state.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

list-before

### After

list-after

## Testing Steps / QA Criteria

1. Select list from nav bar
2. Enter an item into the filter item input
3. List items should be filtered by what user entered
4. Click clear button to clear input and unfilter list

### CODE REVIEW

**Review by kaitlinLewis — APPROVED** [review]
Nice work!

**Comment by NuriyaAkh** [comment]
Woo hoo! Great work!

**Review by NuriyaAkh — APPROVED** [review]

File: src/views/List.jsx [hunk 1]
```diff
@@ -1,15 +1,39 @@
...
+		e.preventDefault();
+		setSearchQuery('');
+	};
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
+			{/* <div> */}
```
Comment (cafloyd) [inline src/views/List.jsx:14]: Please be sure to remove this unneeded comment as well as the closing tag on line 28!

**Review by cafloyd — CHANGES_REQUESTED** [review]
Nice work!! The filter seems to work as expected and will match any string against the shopping list, regardless of where that string falls in the words. The 'clear' button seems to work using keyboard navigation, which is great!

One small change request for code cleanliness, but otherwise looks awesome!

**Review by cafloyd — APPROVED** [review]
🎉

**Review by georgeoffley — APPROVED** [review]
The logic looks good! I would just make the adjustment @cafloyd suggested, and you're golden.

**Review by daredtech — APPROVED** [review]

**Pull request merged**